src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/GeneratorDiagnostic.cs

```diff
@@ -55,8 +57,130 @@ public override DiagnosticInfo ToDiagnosticInfo(DiagnosticDescriptor descriptor,
                     UnnecessaryDataLocations,
                     // Add "unnecessary locations" property so the IDE fades the right locations.
                     DiagnosticProperties.Add(WellKnownDiagnosticTags.Unnecessary, $"[{string.Join(",", Enumerable.Range(0, UnnecessaryDataLocations.Length))}]"),
-                    UnnecessaryDataDetails,
-                    elementName);
+                    UnnecessaryDataName,
+                    elementName,
+                    UnnecessaryDataDetails ?? "");
+            }
+        }
+
+        /// <summary>
+        /// Provides an implementation of <see cref="IMarshallingGenerator.SupportsByValueMarshalKind(ByValueContentsMarshalKind, TypePositionInfo, StubCodeContext, out GeneratorDiagnostic?)"/> through <see cref="GetSupport(ByValueContentsMarshalKind, TypePositionInfo, StubCodeContext, out GeneratorDiagnostic?)"/>
+        /// </summary>
+        public record ByValueMarshalKindSupportManager(
```

Comment:
I think we should separate this out to its own file or at least have it as a top-level type. I want to keep the `GeneratorDiagnostic` type as effectively a discriminated union with the nested types being the different options.

Also, can we have a different name here instead of "Manager"? I think I would prefer `ByValueMarshalKindSupportDescriptor` as it's describing the static data.

Reply:
Good idea, and I agree, I really don't like manager